dexlib2/src/main/java/com/android/tools/smali/dexlib2/writer/pool/PoolClassDef.java

```diff
@@ -30,46 +30,51 @@
+import com.android.tools.smali.util.CollectionUtils;
+import com.android.tools.smali.util.IteratorUtils;
 
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableSortedSet;
-import com.google.common.collect.Iterables;
-import com.google.common.collect.Iterators;
-import com.google.common.collect.Ordering;
 import java.util.AbstractCollection;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Set;
 import java.util.SortedSet;
+import java.util.stream.Collectors;
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
 class PoolClassDef extends BaseTypeReference implements ClassDef {
     @Nonnull final ClassDef classDef;
     @Nonnull final TypeListPool.Key<List<String>> interfaces;
-    @Nonnull final ImmutableSortedSet<Field> staticFields;
-    @Nonnull final ImmutableSortedSet<Field> instanceFields;
-    @Nonnull final ImmutableSortedSet<PoolMethod> directMethods;
-    @Nonnull final ImmutableSortedSet<PoolMethod> virtualMethods;
+    @Nonnull final SortedSet<Field> staticFields;
+    @Nonnull final SortedSet<Field> instanceFields;
+    @Nonnull final SortedSet<PoolMethod> directMethods;
+    @Nonnull final SortedSet<PoolMethod> virtualMethods;
 
     int classDefIndex = DexPool.NO_INDEX;
     int annotationDirectoryOffset = DexPool.NO_OFFSET;
 
     PoolClassDef(@Nonnull ClassDef classDef) {
         this.classDef = classDef;
 
-        interfaces = new TypeListPool.Key<List<String>>(ImmutableList.copyOf(classDef.getInterfaces()));
-        staticFields = ImmutableSortedSet.copyOf(classDef.getStaticFields());
-        instanceFields = ImmutableSortedSet.copyOf(classDef.getInstanceFields());
-        directMethods = ImmutableSortedSet.copyOf(
-                Iterables.transform(classDef.getDirectMethods(), PoolMethod.TRANSFORM));
-        virtualMethods = ImmutableSortedSet.copyOf(
-                Iterables.transform(classDef.getVirtualMethods(), PoolMethod.TRANSFORM));
+        interfaces = new TypeListPool.Key<List<String>>(Collections.unmodifiableList(new ArrayList<>(classDef.getInterfaces())));
+        staticFields = ArraySortedSet.copyOf(CollectionUtils.naturalOrdering(), 
+            IteratorUtils.toList(classDef.getStaticFields()));
+        instanceFields = ArraySortedSet.copyOf(CollectionUtils.naturalOrdering(), 
+            IteratorUtils.toList(classDef.getInstanceFields()));
+        directMethods = ArraySortedSet.copyOf(CollectionUtils.naturalOrdering(),
+                IteratorUtils.toList(classDef.getDirectMethods()).stream().map(PoolMethod.TRANSFORM)
+                        .collect(Collectors.toList()));
+        virtualMethods = ArraySortedSet.of(CollectionUtils.naturalOrdering(),
+                IteratorUtils.toList(classDef.getVirtualMethods()).stream().map(PoolMethod.TRANSFORM)
+                .collect(Collectors.toList()));
```

Comment:
ArraySortedSet is immutable, since it doesn't support edit operations, it's unfortunately not very clear from the name but I added that in its javadoc. Do you still prefer to add the unmodifiable wrapper to make it clearer to read?

Reply:
Thanks